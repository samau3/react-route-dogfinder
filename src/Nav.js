import React from "react";
import { NavLink } from "react-router-dom";

function Nav({ names }) {
    return (
        <ul>
            <NavLink exact to="/dogs">
                Home
            </NavLink>
            {names.map(name => (
                <li key={name}>
                    <NavLink exact to={`/dogs/${name}`}>{name}</NavLink>
                </li>
            ))}
        </ul>
    );
}

export default Nav;
