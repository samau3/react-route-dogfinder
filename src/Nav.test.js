import React from 'react';
import Nav from './Nav';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { TEST_DOGS } from './_testCommon';

// basic tests
it('renders without crashing', function () {
    render(
        <MemoryRouter>
            <Nav dogs={TEST_DOGS} />
        </MemoryRouter>
    );
});

it('matches snapshot', function () {
    const container = render(
        <MemoryRouter>
            <Nav dogs={TEST_DOGS} />
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});
// end basic tests

// full render
it('mounts without crashing', function () {
    const { getByText, container, debug } = render(
        <MemoryRouter>
            <Nav dogs={TEST_DOGS} />
        </MemoryRouter>
    );

    // debug()
    const links = container.querySelectorAll("li a")
    const dogLink = getByText("Whiskey");
    expect(dogLink).toBeInTheDocument();
    expect(links).toHaveLength(3);
});
