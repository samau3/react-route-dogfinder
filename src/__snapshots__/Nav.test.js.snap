// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <ul>
    <a
      href="/dogs"
    >
      Home
    </a>
    <li>
      <a
        href="/dogs/whiskey"
      >
        Whiskey
      </a>
    </li>
    <li>
      <a
        href="/dogs/duke"
      >
        Duke
      </a>
    </li>
    <li>
      <a
        href="/dogs/perry"
      >
        Perry
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
